### PR TITLE
Widget: Is repeat data and pin data slot

### PR DESCRIPTION
**Pull request by rubenberttpingol**
relates to xibosignageltd/xibo-private#371

### CODE REVIEW

**Review by maurofmferrao — CHANGES_REQUESTED** [review]
All looking good, just a minor change! :)

File: modules/src/player.js [hunk 1]
```diff
@@ -694,10 +815,14 @@ $(function() {
                         }
 
                         if (_dataKey >= slot && _dataKey < data?.length) {
-                          const currentSlot = slot + 1;
+                          const currentSlot = slot;
                           const currentKey = _dataKey + 1;
                           const moduloEq = currentSlot === maxSlot ?
                             0 : currentSlot;
+
+                          if (!isPinSlot && currentKey >= maxSlot) {
```
[the remark this answers is not in the document]
Comment (rubenpingol-xibo) [reply modules/src/player.js:823]: Wait, this `renderDataItems` block should actually be removed.

[the remark this answers is not in the document]
Comment (rubenpingol-xibo) [reply modules/src/player.js:823]: This is already replaced by new `renderDataItem`

**Review by maurofmferrao — APPROVED** [review]
Looks good!

**Pull request merged**